modules/machineconfig-modify-dns-nameservers.adoc

```diff
@@ -0,0 +1,174 @@
+// Module included in the following assemblies:
+//
+// * post_installation_configuration/machine-configuration-tasks.adoc
+
+[id="machineconfig-modify-dns-nameservers_{context}"]
+= Modifying the DNS name servers
+
+You can override the DNS name servers and search list in the `/etc/resolv.conf` file.
+Some operating systems use `NetworkManager` to manage network interfaces.
+When `NetworkManager` manages some interfaces, it also manages the contents of the `/etc/resolv.conf` file.
+The following procedure shows how to replace the `/etc/resolv.conf` file and prevent `NetworkManager` from managing the file.
+
+.Procedure
+
+. Create the contents of the `resolv.conf` file and encode it as base64. For example:
++
+[source,terminal]
+----
+$ cat << EOF | base64
```

Comment:
`base64` with no options will cause line breaks every 77 characters to appear when the user copies and pastes it into their machine config which will cause errors.  The user will have to delete the line breaks in order to remedy the errors.  I prefer `base64 -w0` which will not word wrap.  The only caveat is that there is no line break at in the output before the next shell prompt.  You can opt to do `base64 -w99999999999999` which will add a line break at the very end as long as the base64 encoded output is not longer than 99999999999999 characters but this looks a little hacky.

**Example: Without Option**
```
[mnguyen@pet32 ~]$ cat << EOF | base64    
# Custom resolv.conf added with Machine Config Operator
# and the override-etc-resolv-conf.yaml file.
search c.openshift-gce-devel-ci.internal google.internal c.openshift-gce-devel-ci.inte
nameserver 169.254.169.254                                                            
nameserver 8.8.8.8                                                            
EOF                                                                           
IyBDdXN0b20gcmVzb2x2LmNvbmYgYWRkZWQgd2l0aCBNYWNoaW5lIENvbmZpZyBPcGVyYXRvcgoj
IGFuZCB0aGUgb3ZlcnJpZGUtZXRjLXJlc29sdi1jb25mLnlhbWwgZmlsZS4Kc2VhcmNoIGMub3Bl
bnNoaWZ0LWdjZS1kZXZlbC1jaS5pbnRlcm5hbCBnb29nbGUuaW50ZXJuYWwgYy5vcGVuc2hpZnQt
Z2NlLWRldmVsLWNpLmludGUKbmFtZXNlcnZlciAxNjkuMjU0LjE2OS4yNTQKbmFtZXNlcnZlciA4
LjguOC44Cg==
[mnguyen@pet32 ~]$ 
```

**Example: With `-w0` Option**
```
[mnguyen@pet32 ~]$ cat << EOF | base64 -w0
# Custom resolv.conf added with Machine Config Operator
# and the override-etc-resolv-conf.yaml file.
search c.openshift-gce-devel-ci.internal google.internal c.openshift-gce-devel-ci.inte
nameserver 169.254.169.254                                                            
nameserver 8.8.8.8                                                            
EOF                                                                           
IyBDdXN0b20gcmVzb2x2LmNvbmYgYWRkZWQgd2l0aCBNYWNoaW5lIENvbmZpZyBPcGVyYXRvcgojIGFuZCB0aGUgb3ZlcnJpZGUtZXRjLXJlc29sdi1jb25mLnlhbWwgZmlsZS4Kc2VhcmNoIGMub3BlbnNoaWZ0LWdjZS1kZXZlbC1jaS5pbnRlcm5hbCBnb29nbGUuaW50ZXJuYWwgYy5vcGVuc2hpZnQtZ2NlLWRldmVsLWNpLmludGUKbmFtZXNlcnZlciAxNjkuMjU0LjE2OS4yNTQKbmFtZXNlcnZlciA4LjguOC44Cg==[mnguyen@pet32 ~]$ 
```

Reply:
+1 to you,@mike-nguyen.